security/pkg/nodeagent/caclient/providers/citadel/client_test.go

```diff
@@ -107,3 +115,99 @@ func TestCitadelClient(t *testing.T) {
 		}
 	}
 }
+
+type mockTokenCAServer struct {
+	Certs []string
+	Err   error
+}
+
+func (ca *mockTokenCAServer) CreateCertificate(ctx context.Context, in *pb.IstioCertificateRequest) (*pb.IstioCertificateResponse, error) {
+	client := fake.NewSimpleClientset()
+	tokenReview := &k8sauth.TokenReview{
+		Spec: k8sauth.TokenReviewSpec{
+			Token: validToken,
+		},
+	}
+	tokenReview.Status.Audiences = []string{}
+	tokenReview.Status.Authenticated = true
+	tokenReview.Status.User = k8sauth.UserInfo{
+		Username: "system:serviceaccount:default:example-pod-sa",
+		Groups:   []string{"system:serviceaccounts"},
+	}
+	client.PrependReactor("create", "tokenreviews", func(action ktesting.Action) (bool, runtime.Object, error) {
+		return true, tokenReview, nil
+	})
+	authenticator := authenticate.NewKubeJWTAuthenticator(client, "Kubernetes", nil, "example.com", jwt.PolicyFirstParty)
+	_, err := authenticator.Authenticate(ctx)
+	if err == nil {
+		return &pb.IstioCertificateResponse{CertChain: ca.Certs}, nil
+	}
+
+	return nil, err
+}
+
+func TestCitadelClientWithDifferentTypeToken(t *testing.T) {
+	testCases := map[string]struct {
+		server       mockTokenCAServer
+		expectedCert []string
+		expectedErr  string
+		token        string
+	}{
+		"Valid Token": {
+			server:       mockTokenCAServer{Certs: fakeCert, Err: nil},
+			expectedCert: fakeCert,
+			expectedErr:  "",
+			token:        validToken,
```

Comment:
the token is not used in the test, in fact if you replace this `validToken` with the `fakeToken`, the test is still passing.

Reply:
I am refactor the code remove useless and redundant code